# Do people believe that men make better political leaders than women?

* Q29 - Men make better political leaders than women do
______________________________________________________

* Q260 – Gender of the respondent (1 = Male, 2 = Female) 
* Q261 – Age of respondent (Numeric)
* Q275A – Education level (Country-specific categories)
* Q50 – Satisfaction with financial situation (Scale from 1 to 10)
* Q42 – Attitudes toward societal change (People who support radical or conservative views may have different attitudes towards women in politics.)
* Q71 – Confidence in government (Measures trust in political institutions)
* Q30 – Is university education more important for boys than girls? (Binary)
* Q33 – Should men have more right to jobs than women when jobs are scarce?
* Q6 – How important is religion in life? (Scale from 1 to 4)
* Q266 - Country of birth


In [1]:
import pandas as pd
import numpy as np

In [8]:
df= pd.read_csv("C:\\nis\DA3\WVS_subset.csv")
df

B_COUNTRY B_COUNTRY_ALPHA  C_COW_NUM C_COW_ALPHA  D_INTERVIEW  A_YEAR  \
0             20             AND        232         AND     20070001    2018   
1             20             AND        232         AND     20070002    2018   
2             20             AND        232         AND     20070003    2018   
3             20             AND        232         AND     20070004    2018   
4             20             AND        232         AND     20070005    2018   
...          ...             ...        ...         ...          ...     ...   
97215        716             ZWE        552         ZIM    716071211    2020   
97216        716             ZWE        552         ZIM    716071212    2020   
97217        716             ZWE        552         ZIM    716071213    2020   
97218        716             ZWE        552         ZIM    716071214    2020   
97219        716             ZWE        552         ZIM    716071215    2020   

          S025  A_STUDY  A_WAVE  W_WEIGHT  ...  Q283  Q284  Q285  Q286  Q287  \
0       202018        2       7       1.0  ...     9     2     2     3     3   
1       202018        2       7       1.0  ...     4     2     1     1     2   
2       202018        2       7       1.0  ...     5     2     1     2     3   
3       202018        2       7       1.0  ...     9     2     2     2     4   
4       202018        2       7       1.0  ...     8     2     1     1     4   
...        ...      ...     ...       ...  ...   ...   ...   ...   ...   ...   
97215  7162020        2       7       1.0  ...     0    -3     2     2     3   
97216  7162020        2       7       1.0  ...     1     3     1     1     2   
97217  7162020        2       7       1.0  ...     0     2     2     2     4   
97218  7162020        2       7       1.0  ...     6     2     1     2     4   
97219  7162020        2       7       1.0  ...     0     3     1     2     4   

       Q288  Q288R  Q289    Q289CS9    Q290  
0         5      2     1   10100000   20001  
1         9      3     1   10100000   20001  
2         5      2     1   10100000   20001  
3         4      2     0  100000020   20001  
4         4      2     0  100000020   20001  
...     ...    ...   ...        ...     ...  
97215     5      2     1   10100000  716006  
97216     5      2     2   20000000  716007  
97217     5      2     1   10100000  716006  
97218     1      1     1   10100000  716006  
97219     1      1     1   10100000  716006  

[97220 rows x 168 columns]

In [16]:
selected_columns = [
    "Q29", 
    "Q260", "Q261", "Q275A", "Q50", "Q71", 
    "Q30", "Q33", "Q6", "Q266", "Q42"
]

df_filtered = df[selected_columns]

df_filtered


Q29  Q260  Q261  Q275A  Q50  Q71  Q30  Q33  Q6  Q266  Q42
0        3     2  1958  20003    5    1    3    4   4   724    2
1        3     1  1971  20007    9    4    3    4   4    20    2
2        3     1  1969  20007    8    3    3    4   3   724    2
3        4     2  1956  20002    6    3    4    4   4   724    1
4        3     1  1969  20002    7    2    3    4   3   724    1
...    ...   ...   ...    ...  ...  ...  ...  ...  ..   ...  ...
97215    1     2  1977     -4    1    1    3    1   2   716    2
97216    1     1  1986     -4    5    4    4    5   1   716    2
97217    3     2  1998     -4    5    2    3    4   2   716    3
97218    3     2  1972     -4    1    2    3    4   1   716    1
97219    3     1  1982     -4    1    3    3    2   1   716    3

[97220 rows x 11 columns]

In [17]:
df_filtered = df_filtered[(df_filtered >= 0).all(axis=1)]
df_filtered

Q29  Q260  Q261   Q275A  Q50  Q71  Q30  Q33  Q6  Q266  Q42
0        3     2  1958   20003    5    1    3    4   4   724    2
1        3     1  1971   20007    9    4    3    4   4    20    2
2        3     1  1969   20007    8    3    3    4   3   724    2
3        4     2  1956   20002    6    3    4    4   4   724    1
4        3     1  1969   20002    7    2    3    4   3   724    1
...    ...   ...   ...     ...  ...  ...  ...  ...  ..   ...  ...
93605    1     1  1995  860007   10    1    1    2   1   860    2
93606    2     2  1968  860007    7    3    1    2   1   860    2
93607    2     1  1964  860007   10    1    1    3   2   860    1
93612    3     2  1989  860007   10    1    1    1   1   860    2
93614    3     2  1981  860005    7    2    3    1   2   860    2

[47143 rows x 11 columns]

In [19]:
important_columns = ["Q29", "Q260", "Q261", "Q42", "Q71"]
round(df_filtered[important_columns].describe(), 2)

Q29      Q260      Q261       Q42       Q71
count  47143.00  47143.00  47143.00  47143.00  47143.00
mean       2.61      1.52   1975.40      1.95      2.61
std        0.96      0.50     16.37      0.58      0.98
min        1.00      1.00   1916.00      1.00      1.00
25%        2.00      1.00   1963.00      2.00      2.00
50%        3.00      2.00   1977.00      2.00      3.00
75%        3.00      2.00   1989.00      2.00      3.00
max        4.00      2.00   2007.00      3.00      4.00

* Q29 (Men better political leaders?) – Mean 2.61, indicating a lean towards disagreement but with variation (std = 0.96).
* Q260 (Gender) – Balanced sample (Mean = 1.52, close to equal split between men and women).
* Q261 (Age, Birth Year) – Respondents born mostly between 1963-1989 (Median = 1977).
* Q42 (Attitude toward societal change) – Median 2.00, suggesting most favor gradual reforms over radical change or strict conservatism.
* Q71 (Confidence in Government) – Mean 2.61, implying moderate confidence, but some polarization (std = 0.98).